### PR TITLE
connectedhomeip: samples: Added light bulb and light switch samples.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
* Added light bulb sample
* Added light switch sample
* Added support for nRF5340
* Updated CHIP revision
* Fixed a couple of issues in the CHIP samples:
  * switched to Oberon SHA256 to workaround a hardware limitation (CC310 unable to read flash memory)
  * erasing Thread configuration on init not to leave a device in a partially persisted state (CHIP doesn't support persistent storage yet)
  * removed spaces from NFC URI Tag

To reviewers: files in the "gen/" directories are auto-generated using the [ZAP](https://github.com/project-chip/zap) tool, so detailed review of their contents probably doesn't make much sense.

### CODE REVIEW

**Comment by kkasperczyk-no** [comment]
Do not remove DNM label until sdk-connectedhomeip version will be updated.

**Comment by doublemis1** [comment]
sample.yamls looks fine

File: samples/connectedhomeip/light_bulb/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+get_filename_component(CHIP_ROOT ../../../../modules/lib/connectedhomeip REALPATH)
+get_filename_component(COMMON_ROOT ../common REALPATH)
+
+list(APPEND ZEPHYR_EXTRA_MODULES ${CHIP_ROOT}/config/nrfconnect/chip-module)
+find_package(Zephyr HINTS $ENV{ZEPHYR_BASE})
+
+project(chip-light-bulb)
+
+# Temporary get rid of compilation warnings till zephyrproject-rtos/zephyr/pull/31471 is fetched
+zephyr_compile_options(-Wno-parentheses)
```
Comment (LuDuda) [inline samples/connectedhomeip/light_bulb/CMakeLists.txt:18]: This can be removed now.

File: samples/connectedhomeip/light_bulb/README.rst [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+.. |sample path| replace:: :file:`samples/connectedhomeip/light_bulb`
+
+.. include:: /includes/build_and_run.txt
+
+.. _chip_light_bulb_building_with_light_switch_overlay:
+
+Building for Light switch test mode
+===================================
+
+To build the sample with configuration that allows testing with light switch sample, run the following command:
```
Comment (LuDuda) [inline samples/connectedhomeip/light_bulb/README.rst:114]: Should we mention somewhere, what is the test mode? I.e. that Commissioning process is bypassed, and keys hardcoded.

File: samples/connectedhomeip/light_bulb/src/chip_project_config.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+#pragma once
+
+/* Use a default pairing code if one hasn't been provisioned in flash. */
+#define CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE 12345678
+#define CHIP_DEVICE_CONFIG_USE_TEST_SETUP_DISCRIMINATOR 0xF00
+
+#define LIGHT_BULB_PWM_DEVICE	DT_PWMS_LABEL(DT_ALIAS(pwm_led1))
+#define LIGHT_BULB_PWM_CHANNEL	DT_PWMS_CHANNEL(DT_ALIAS(pwm_led1))
```
Comment (LuDuda) [inline samples/connectedhomeip/light_bulb/src/chip_project_config.h:23]: nit: alignment

File: samples/connectedhomeip/light_bulb/src/main.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		return ret;
+	}
+
+	ret = ThreadStackMgr().InitThreadStack();
+	if (ret != CHIP_NO_ERROR) {
+		LOG_ERR("ThreadStackMgr().InitThreadStack() failed");
+		return ret;
+	}
+
+	/* Until all CHIP data needed to perform full communication are not saved in the persistent storage,
```
Comment (LuDuda) [inline samples/connectedhomeip/light_bulb/src/main.cpp:45]: Can we add it to the docuemntation as well for both examples?

File: samples/connectedhomeip/light_bulb/src/light_bulb_publish_service.cpp [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+#include <inet/InetConfig.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <transport/TransportMgr.h>
+#include <transport/raw/UDP.h>
+
+LOG_MODULE_DECLARE(app);
+
+const char kPublishServicePayload[] = "chip-light-bulb";
+const char kPublishDestAddr[] = "ff02::1";
```
Comment (LuDuda) [inline samples/connectedhomeip/light_bulb/src/light_bulb_publish_service.cpp:22]: Thi should be changed to `ff03::1`. After that, we should also update the doc, changing `fe80..` to `fd..`.

File: samples/connectedhomeip/light_bulb/README.rst (hunk 2 above)
Comment (greg-fer) [reply samples/connectedhomeip/light_bulb/README.rst]: I think this is already covered by the reused lines from the door lock sample:

Remote testing using test mode
==============================

.. include:: ../lock/README.rst
    :start-after: chip_door_lock_sample_test_mode_start
    :end-before: chip_door_lock_sample_test_mode_end

Which expands to:

Alternatively to the commissioning procedure, you can use the test mode, which allows to join a Thread network with default static parameters and static cryptographic keys. Use Button 3 to enable this mode after building and running the sample.

**Review by greg-fer — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Review by doublemis1 — APPROVED** [review]

**Comment by rlubos** [comment]
@LuDuda Why did you remove DNM label? The manifest file still  points to a pull request.

**Review by rlubos — COMMENTED** [review]
LGTM, pretty neat. I can approve after the manifest is updated with proper SHA.

**Comment by Damian-Nordic** [comment]
@rlubos I updated the manifest with the final SHA. We were not sure if we should first wait for review and then merge changes to sdk-connectedhomeip or the other way around.

**Review by rlubos — APPROVED** [review]

**Pull request merged**